.github/workflows/build-push-pr.yml

```diff
@@ -0,0 +1,34 @@
+name: Build container image for PR
+on:
+  pull_request:
+    types: [labeled, synchronize, reopened, ready_for_review, opened]
+
+permissions:
+  contents: read
+  packages: write
+
+jobs:
+  compute-suffix:
+    runs-on: ubuntu-latest
+    if: ${{ !github.event.pull_request.draft && contains(github.event.pull_request.labels.*.name, 'build-image') }}
+    steps:
+      - id: version_vars
+        run: |
+          echo mastodon_version_suffix=+pr-${{ github.event.pull_request.number }}-$(git rev-parse --short HEAD) >> $GITHUB_OUTPUT
+    outputs:
+      suffix: ${{ steps.version_vars.outputs.mastodon_version_suffix }}
+
+  build-image:
+    needs: compute-suffix
+    uses: ./.github/workflows/build-container-image.yml
+    with:
+      platforms: linux/amd64,linux/arm64
+      use_native_arm64_builder: true
```

Comment:
Do we want native ARM64 builds for these PRs?

Reply:
I added it as it should now be fast with the native builder, and it allows to manually trigger a full Docker build for any PR, which might be useful to test if it really works when doing some build changes

Reply:
Sounds good, although that means we need to restrict who can set those labels.

Reply:
Labels can be set by people in the organisation, so triage team + devs. I think it mitigates sufficiently the issue.